homeassistant/components/sentry/__init__.py

```diff
@@ -44,15 +51,98 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+            "channel": channel,
+            "installation_type": system_info["installation_type"],
+            "uuid": huuid,
+        }
+
+        # Find out all integrations in use, filter "auth", because it
+        # triggers security rules, hiding all data.
+        integrations = [
+            integration
+            for integration in hass.config.components
+            if integration != "auth" and "." not in integration
+        ]
+
+        # Additional extra data to add to the event
+        extra_data = {
+            **system_info,
+            "custom_components": "\n".join(sorted(custom_components)),
+            "integrations": "\n".join(sorted(integrations)),
+        }
+
+        # Add additional tags based on what caused the event.
+        platform = entity_platform.current_platform.get()
+        if platform is not None:
+            # This event happened in a platform
+            additional_tags["custom_component"] = "no"
+            additional_tags["integration"] = platform.platform_name
+            additional_tags["platform"] = platform.domain
+        elif "logger" in event:
+            # Logger event, try to get integration information from the logger name.
+            matches = LOGGER_INFO_REGEX.findall(event["logger"])
+            if matches:
+                group1, group2, group3, group4 = matches[0]
+                # Handle the "homeassistant." package differently
+                if group1 == "homeassistant" and group2 and group3:
+                    if group2 == "components":
+                        # This logger is from a component
+                        additional_tags["custom_component"] = "no"
+                        additional_tags["integration"] = group3
+                        if group4 and group4 in ENTITY_COMPONENTS:
+                            additional_tags["platform"] = group4
+                    else:
+                        # Not a component, could be helper, or something else.
+                        additional_tags[group2] = group3
+                else:
+                    # Not the "homeassistant" package, this third-party
+                    additional_tags["package"] = group1
+
+        # If this event is caused by an integration, add a tag if this
+        # integration is custom or not.
+        if (
+            "integration" in additional_tags
+            and additional_tags["integration"] in custom_components
+        ):
+            additional_tags["custom_component"] = "yes"
+
+        # Update event with all additional data
+        event.setdefault("tags", {}).update(additional_tags)
+        event.setdefault("extra", {}).update(extra_data)
+
+        return event
+
     sentry_sdk.init(
         dsn=conf.get(CONF_DSN),
         environment=conf.get(CONF_ENVIRONMENT),
-        integrations=[sentry_logging],
-        release=f"homeassistant-{__version__}",
+        integrations=[sentry_logging, AioHttpIntegration(), SqlalchemyIntegration()],
```

Comment:
there is a way to tell sentry to "automatically enable whatever integrations it can" - may be useful for hass to future proof (though still coupled to the sdk version)

the one caveat to it is it wont enable any dangerous integrations (ones which might not be desired generally or have side effects)

Reply:
You can see an example of it here in the init call:

https://docs.sentry.io/product/performance/getting-started/?platform=python

```
_experiments={"auto_enabling_integrations": True}
```

(going to chat internally about if/when we plan to turn this on by default, or graduate it from 'experiments')

Reply:
As long as it is experimental, lets not. Furthermore, from all supported things by Sentry, this is what we can use.